src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs

```diff
@@ -203,15 +208,25 @@ private async Task UpdateJobRecordAsync(CancellationToken cancellationToken)
                 SearchResult searchResult = null;
 
                 // Search and process the results.
-                using (IScoped<ISearchService> searchService = _searchServiceFactory())
+                switch (_exportJobRecord.ExportType)
                 {
-                    searchResult = await searchService.Value.SearchAsync(
-                        resourceType: null,
-                        queryParametersList,
-                        cancellationToken);
+                    case ExportJobType.All:
+                    case ExportJobType.Patient:
+                        using (IScoped<ISearchService> searchService = _searchServiceFactory())
+                        {
+                            searchResult = await searchService.Value.SearchAsync(
+                                resourceType: null,
+                                queryParametersList,
+                                cancellationToken);
+                        }
+
+                        break;
+                    case ExportJobType.Group:
+                        searchResult = await GetGroupPatients(_exportJobRecord.GroupId, queryParametersList, cancellationToken);
```

Comment:
Once we have a group of patients and we run compartment search on each of them. If some of these patients share some common resources, say a `Practitioner`, we will end up exporting duplicates of that resource.

Reply:
There is a separate story for dealing with duplicate resources [74403](https://microsofthealth.visualstudio.com/Health/_workitems/edit/74403)